internal/services/pubkey_service.go

```diff
@@ -99,6 +109,50 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkey, err := dao.GetPubkeyDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err))
+		return
+	}
+
+	resources, err := pubkey.UnscopedListByPubkeyId(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			arn, errArn := sourcesClient.GetAuthentication(r.Context(), res.SourceID)
+			if errArn != nil {
+				if errors.Is(err, pkgErr.ApplicationNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: application not found", errArn, 404))
+					return
+				}
+				if errors.Is(err, pkgErr.AuthenticationForSourcesNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: authentication not found", errArn, 404))
+					return
+				}
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources", errArn, 500))
+				return
+			}
+
+			ec2Client, errEc2 := clients.GetCustomerEC2Client(r.Context(), arn, "us-east-1")
```

Comment:
we can now use the region from the resouce I believe:
```suggestion
			ec2Client, errEc2 := clients.GetCustomerEC2Client(r.Context(), arn, res.Region)
```